### PR TITLE
Security jwt

### PR DESCRIPTION
**Pull request by SerhiiAN**

### CODE REVIEW

File: src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java]: save it to constants

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            UserDetails userDetails = userDetailsService.loadUserByUsername(userName);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);

```

File: src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String TOKEN_STARTS = "Bearer ";
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java]:
```suggestion
    private static final String TOKEN_HEADER = "Bearer ";

```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
some comments

File: src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_HEADER)) {
+            return bearerToken.substring(7);
```
Comment (okuzan) [inline src/main/java/mate/academy/intro/security/JwtAuthenticationFilter.java]:
```suggestion
        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_HEADER)) {
            return bearerToken.substring(TOKEN_HEADER.length());
```
that's where we get 7 from, code should reflect this

**Review by okuzan — CHANGES_REQUESTED** [review]
Almost there!

File: src/main/java/mate/academy/intro/dto/user/UserLoginRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.intro.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @Email
+        @Size(min = 8, max = 35)
+        @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/dto/user/UserLoginRequestDto.java]:
```suggestion
        @NotBlank
```
For String

File: src/main/java/mate/academy/intro/dto/user/UserLoginRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @Email
+        @Size(min = 8, max = 35)
+        @NotEmpty
+        String email,
+        @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/dto/user/UserLoginRequestDto.java]: same

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Well done. See comment

**Review by sarakhmen — APPROVED** [review]
lgtm!

**Pull request merged**